integtest/spec/helper/source.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require_relative 'book'
+require_relative 'repo'
+
+##
+# Helper class for setting up source files for tests.
+class Source
+  attr_reader :books
+
+  def initialize(tmp)
+    @root = File.expand_path 'src', tmp
+    Dir.mkdir @root
+    @repos = []
+    @books = []
+  end
+
+  ##
+  # Create a source repo to which you can add files to be converted.
+  def repo(name)
+    Repo.new(name, path(name)).tap { |r| @repos.push r }
```

Comment:
why would you just write
`@repos << Repo.new(name, path(name))` ?

Reply:
:man_facepalming: I think it made more sense in a previous iteration....

Reply:
It turns out I need to return the new repo as well as add it which is why I did this. Could you suggest something that makes that more obvious? I'll push some comments too.

Reply:
The three ways that come to mind are:
```ruby
@repos << Repo.new(name, path(name)) && @repos[-1]
```

```ruby
repo = Repo.new(name, path(name))
(@repos << repo) && repo
```

```ruby
repo = Repo.new(name, path(name))
@repos << repo
repo
```

The use of `#tap` is certainly valid but seems not that necessary.